### ISSUE TITLE
GoCardless not working

### ISSUE THREAD

**Issue by IdaEdz** [FrancisRalph/actual-budget-gcp-cloud-run#1]
Hey I based my installation off your repository.
I'm wondering if GoCardless is working for you?

After I configure GoCardless with ID and secret,
I try to add my bank. I choose any country and wait for the site to load /gocardless/get-banks
~~But the server never returns with a respons.~~ Resturns with 200 after ~1-2min with a internal error.

I followed Actuals guide here: https://actualbudget.org/docs/advanced/bank-sync/gocardless/
But it might not work with Cloud Run?

There is a github repo about adding new banks on the normal server version of Actual here:
https://github.com/actualbudget/actual-server/blob/master/src/app-gocardless/README.md

But I don't see how it would be possible for me to configure that.

**Comment by FrancisRalph** [comment]
When the server does not respond to get-banks, could you check the logs from your Cloud Run service?

Unfortunately my country is not supported by GoCardless so I've not used it. But I can help you debug.

**Comment by IdaEdz** [comment]
Logfile from google when accessing /get-banks

[sanitized-logfile.json](https://github.com/user-attachments/files/17591313/sanitized-logfile.json)

**Comment by FrancisRalph** [comment]
Hmm, looked into Actual's source code and nothing wrong on their end. It's just GoCardless returning a really vague INTERNAL_ERROR, but at the same time responding with a status code of 200 (so actual doesnt know it's an error). Try again, if you're facing the same issue you'll probably have to contact GoCardless support.

Specifically, it's the getInstitutions API that's returning the error you're getting: https://github.com/actualbudget/actual-server/blob/b5f8aa4d0549b8380b717df11882e6c0213bbb09/src/app-gocardless/app-gocardless.js#L96-L117

**Comment by IdaEdz** [comment]
Hey,
I'm trying to interpret this page from the actual docs:
[https://actualbudget.org/docs/advanced/bank-sync](https://actualbudget.org/docs/advanced/bank-sync)

"You will need to add a config file to your installation."

I have no clue if that is possible or how to do it.

I know you said that your country is not supported - did you mean that you can't even setup an account and generate a access token? I'd like someone to verify if they can get the list of banks for *any* country in actual to show up.

**Comment by IdaEdz** [comment]
This accually seams to be the main issue:

```
Error /gocardless/get-banks GenericGoCardlessError: GoCardless returned error
at handleGoCardlessError (file:///app/src/app-gocardless/services/gocardless-service.js:60:13)
at Object.setToken (file:///app/src/app-gocardless/services/gocardless-service.js:96:9)
at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
at async file:///app/src/app-gocardless/app-gocardless.js:101:5 {
details: h [Error]: connect ETIMEDOUT 34.117.36.2:443
at h.from (file:///app/node_modules/nordigen-node/dist/index.esm.js:1:5978)
at F.<anonymous> (file:///app/node_modules/nordigen-node/dist/index.esm.js:17:16463)
at F.emit (node:events:517:28)
at Ze.<computed> (file:///app/node_modules/nordigen-node/dist/index.esm.js:17:670)
at ClientRequest.emit (node:events:517:28)
at TLSSocket.socketErrorListener (node:_http_client:501:9)
at TLSSocket.emit (node:events:517:28)
at emitErrorNT (node:internal/streams/destroy:151:8)
at emitErrorCloseNT (node:internal/streams/destroy:116:3)
at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
port: 443,
address: '34.117.36.2',
syscall: 'connect',
code: 'ETIMEDOUT',
errno: -110,
```

The guys over at Actual-discord told me to look at the server's network access.
It feels strange that it would choose to connect "home" to 34.117.36.2:443 as I have not specified that address anywhere.

**Comment by FrancisRalph** [comment]
Hey, yes I can't create an account at all.

I'm also not sure what the config file part of the docs mean. But you can add one in your Cloud Storage bucket /data.

By "home", what do you mean?

**Comment by IdaEdz** [comment]
> By \"home\", what do you mean?

I mean that GoCardless function probably does something to "ping back home" to the server in the Cloud Run.

I believe that whatever the function in the API that tells GoCardless to reply use a call for a hostname that doesn't work well for Cloud Runs.

I would probably have to make a change and specify that to get the DNS address instead.

**Comment by FrancisRalph** [comment]
Hey, I found the issue. The Cloud Run service is running inside a private VPC subnet without a Cloud NAT, which is required for internet access. However, Cloud NAT is a paid product with no free tier.

I placed the Cloud Run service in a VPC as I wanted it to access Cloud Storage through GCP's network instead of through the internet. I now realise I should not put it in a VPC to allow for internet access.

What you can do is remove the "vpc_access" block in main.tf.

**Comment by FrancisRalph** [comment]
Fixed in 42b6c25915096ecf5c3386917e091cc2b81c16ad

Pull the changes, ensure you're authenticated using `gcloud auth application-default login` and run `terraform apply`

**Comment by OskyEdz** [comment]
I ran into a problem trying to run the project with the changes:

```
module.gcs_buckets.google_storage_bucket.buckets["budget.peacehult.se"]: Creating...
╷
│ Error: googleapi: Error 409: Your previous request to create the named bucket succeeded and you already own it., conflict
│
│   with module.gcs_buckets.google_storage_bucket.buckets["budget.peacehult.se"],
│   on .terraform\modules\gcs_buckets\main.tf line 40, in resource "google_storage_bucket" "buckets":
│   40: resource "google_storage_bucket" "buckets" {
│
╵
```
